@xen-orchestra/lite/src/components/Tree.tsx

```diff
@@ -67,18 +71,27 @@ interface Effects {
 interface Computed {}
 
 // Inspired by https://mui.com/components/tree-view/#contentcomponent-prop.
-const CustomContent = React.forwardRef(function CustomContent(props, ref) {
-  const { classes, className, label, nodeId, expansionIcon } = props
+const CustomContent = React.forwardRef(function CustomContent(props: CustomContentProps, ref) {
+  const { classes, className, label, expansionIcon, nodeId, to } = props
   const { handleExpansion, handleSelection, selected } = useTreeItem(nodeId)
 
-  return (
+  return expansionIcon === undefined ? (
+    <Link decorated={false} to={to}>
```

Comment:
Would it be more maintainable/robust to have a function that both changes the URL and calls `handleSelection` instead of having multiple click listeners?